src/plugins/graph/components/graph-wrapper-component.tsx

```diff
@@ -29,12 +29,16 @@ export const GraphWrapperComponent: React.FC<ITileProps> = (props) => {
   const content = model.content as IGraphModel;
   const toolbarProps = useToolbarTileApi({ id: model.id, enabled, onRegisterTileApi, onUnregisterTileApi });
 
-  const handleTileLinkRequest = (tableId: string) => {
+  const handleTileLinkRequest = (tileId: string) => {
     if (enabled) {
       const consumerTile = getTileContentById(model.content, model.id);
       const sharedModelManager = getSharedModelManager(model);
       if (sharedModelManager?.isReady) {
-        const providerDataSet = sharedModelManager?.findFirstSharedModelByType(SharedDataSet, tableId);
+        const existingConsumerDataset = sharedModelManager.findFirstSharedModelByType(SharedDataSet);
```

Comment:
This will find the first `SharedDataSet`, not necessarily one that's linked to the current tile. I think you want something like:
```suggestion
        const existingConsumerDataset = getTileSharedModels(tile).find(m => isSharedDataSet(m))
```